tests/components/application-connector/test/compass-runtime-agent/testkit/compassruntimeagentinit/compass.go

```diff
@@ -0,0 +1,77 @@
+package compassruntimeagentinit
+
+import (
+	"github.com/hashicorp/go-multierror"
+	"github.com/kyma-project/kyma/tests/components/application-connector/test/compass-runtime-agent/testkit/compassruntimeagentinit/types"
+)
+
+type compassconfigurator struct {
+	directorClient types.DirectorClient
+	tenant         string
+}
+
+func NewCompassConfigurator(directorClient types.DirectorClient, tenant string) compassconfigurator {
+	return compassconfigurator{
+		directorClient: directorClient,
+		tenant:         tenant,
+	}
+}
+
+func (cc compassconfigurator) Do(runtimeName, formationName string) (types.CompassRuntimeAgentConfig, types.RollbackFunc, error) {
+	runtimeID, err := cc.directorClient.RegisterRuntime(runtimeName)
+	if err != nil {
+		return types.CompassRuntimeAgentConfig{}, nil, err
+	}
+
+	rollBackFunc := cc.getRollbackFunction(runtimeID, formationName)
+
+	err = cc.directorClient.RegisterFormation(formationName)
+	if err != nil {
+		return types.CompassRuntimeAgentConfig{}, rollBackFunc, err
+	}
+
+	err = cc.directorClient.AssignRuntimeToFormation(runtimeID, formationName)
+	if err != nil {
+		return types.CompassRuntimeAgentConfig{}, rollBackFunc, err
+	}
+
+	token, compassConnectorUrl, err := cc.directorClient.GetConnectionToken(runtimeID)
+	if err != nil {
+		var result *multierror.Error
+
+		multierror.Append(result, err)
+		{
+			err := rollBackFunc()
+			if err != nil {
+				multierror.Append(result, err)
+			}
+		}
```

Comment:
Why is `rollBackFunc` called in this place?

Reply:
The idea was to clean up `Runtime` and `Formation` in case getting token failed. Now I see its redundant, as there is this piece of code in Compass Runtime Agent Configurator
```
compassRuntimeAgentConfig, compassConfiguratorRollbackFunc, err := crc.compassConfigurator.Do(runtimeName, formationName)
	if err != nil {
		return nil, crc.rollbackOnError(err,
			compassConfiguratorRollbackFunc)
	}
```